pylint/checkers/classes/special_methods_checker.py

```diff
@@ -296,9 +296,6 @@ def _is_dict(node: InferenceResult) -> bool:
 
     @staticmethod
     def _is_iterator(node: InferenceResult) -> bool:
-        if node is astroid.Uninferable:
-            # Just ignore Uninferable objects.
-            return True
         if isinstance(node, bases.Generator):
```

Comment:
This kind of safety should not be removed just because we can't cover it. It can be something we added after a crash where someone opened an issue and someone made a hotfix (so lots of work came into it). It's hard to know when exactly astroid is going to not be able to infer something and it's very hard to prove we can remove this kind of thing safely, let's play it safe if we're not sure.

Reply:
This PR is still in draft so I didn't get a chance to explain why I did this. If you inspect the code closely, we're never going to hit this. This is because when `_check_iter` is called within `visit_functiondef`, we already do a truthy check for

```
if (
            inferred
...
```

I checked and `astroid. Uninferable` is falsy. So I think we either keep the removal as I have it or remove the `if inferred` check. This is why it wasn't tested.